js/search.js

```diff
@@ -962,11 +961,39 @@
 				if (id === 'hiddenpowergrass') isViable = (moves.indexOf('energyball') < 0 && moves.indexOf('gigadrain') < 0);
 				if (id === 'hiddenpowerice') isViable = (moves.indexOf('icebeam') < 0 && template.id !== 'xerneas');
 				if (id === 'icywind') isViable = (toId(set.species).substr(0, 6) === 'keldeo');
+				if (id === 'leechlife') isViable = (this.gen > 6);
 				if (id === 'reflecttype') isViable = (toId(set.species) in {latias:1, starmie:1});
 				if (id === 'selfdestruct') isViable = (this.gen < 5 && moves.indexOf('explosion') < 0);
 				if (id === 'skyattack') isViable = (toId(set.species) === 'hawlucha');
 				if (id === 'stunspore') isViable = (moves.indexOf('thunderwave') < 0);
-				if (id === 'toxic') isViable = (this.gen > 1);
+				if (this.gen === 1) {
+					// Usually viable for Gen 1
+					if (id === 'acidarmor' || id === 'amnesia' || id === 'barrier' || id === 'bind' || id === 'clamp' || id === 'confuseray' || id === 'counter' || id === 'firespin' || id === 'hyperbeam' || id === 'mirrormove' || id === 'pinmissile' || id === 'razorleaf' || id === 'sing' || id === 'slash' || id === 'sludge' || id === 'twineedle' || id === 'wrap') isViable = true;
+
+					// Usually not viable for Gen 1
+					if (id === 'disable' || id === 'firepunch' || id === 'icepunch' || id === 'jumpkick' || id === 'leechseed' || id === 'quickattack' || id === 'roar' || id === 'thunderpunch' || id === 'toxic' || id === 'triattack' || id === 'whirlwind') isViable = false;
+
+					// Viable only when certain moves aren't present
+					if (id === 'doubleedge') isViable = (moves.indexOf('bodyslam') < 0);
+					if (id === 'hypnosis') isViable = (moves.indexOf('sleeppowder') < 0);
```

Comment:
Well, with Hypnosis vs Sleep Powder, Hypnosis's poor accuracy [60%, except in DP (modified mid-gen so it's 75% in diamond and pearl but 60% in all other gen 4 games)] means that outside of gen 1 / 3, where sleep isn't as powerful, it's usually not listed as viable (idk about gen 7 though), whereas Sleep Powder is considered viable in any generation.

In later generations it might be more viable to run both Body Slam and Double Edge on the same set (maybe Tauros in GSC or ADV or something in lower tiers?) or equally it's possible that either both are viable separate options (normal-type CBers, Snorlax, etc.) in later generations, or that neither is viable (is venusaur going to run either of those in gen 4 onwards? absolutely not). In ADV, Body Slam is viable on some Pokemon (like Jirachi in OU) but Double Edge would be entirely unviable even if Jirachi didn't get Body Slam.